### DISCUSSION TITLE
Do we have some way to create a custom named command?

### DISCUSSION THREAD

**Discussion by danillos** [helix-editor/helix#6361] [Q&A] [answered]
So I created a Keybinding like:

```
[keys.normal]
F5 = ['trim_selections', ':pipe ./format_json']
```

That converts a selected text like:

```
{"name": "John", "age": 30, "city": "New York"}
```

into:

```
{
  "name": "John",
  "age": 30,
  "city": "New York"
}
```

I would like to instead of press `f5` I just go to `command_pallete` and type `format_json` or type `:format_json`, something like:

```
[commands]
format_json = ['trim_selections', ':pipe ./format_json']
```

With that I don't need to remember all keybindings, I just need to type the command name.
It would be helpful to many commands like `convert_to_snake_case`, `convert_to_camel_case`, ... it opens many new possibilities.

We could even re-use them like:

```
[commands]
format_json = ['trim_selections', ':pipe ./format_json']
format_json_and_send_to_gist = ['fomat_json', ':sh ./send_to_gist']

[keys.normal]
F5 = 'format_json'
F6 = 'format_json_and_send_to_gist'
```

I guess it could be really good to build custom motions too.

**Comment by danillos** [comment]
I just found this another thread: https://github.com/helix-editor/helix/discussions/4692 that seems related to it.

**Comment by the-mikedavis** [comment, marked as answer]
Not currently. This is covered by https://github.com/helix-editor/helix/issues/4423
